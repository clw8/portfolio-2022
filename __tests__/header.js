import React from "react"
import {fireEvent, render } from '@testing-library/react';

import Header from "../src/components/header"

describe("Header", () => {
  let header;
  const onMouseEnter = () => {}
  const onMouseOut = () => {}

  beforeAll(() => {
    header = render(<Header {...{onMouseEnter, onMouseOut, show: true}} />)
  })

  it("renders correctly", () => {
    const firstRender = header.asFragment()
    
    expect(firstRender).toMatchSnapshot()
  })
  
  it("renders hover effects correctly", async() => {
    setTimeout(async() => {
      const projectsLink = await header.findByRole("link", {name: "Projects"});
      fireEvent.mouseEnter(projectsLink);
      expect(onMouseEnter).toHaveBeenCalledTimes(1)
    }, 1000)
  })
})