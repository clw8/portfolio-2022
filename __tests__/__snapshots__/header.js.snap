// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header
    class="react-reveal"
    style="opacity: 1;"
  >
    <div
      class="profile-image"
    >
      <img
        alt="A picture of clw8, the developer of this portfolio site :)"
        height="100"
        src="test-file-stub"
        width="100"
      />
    </div>
    <nav>
      <a
        href="#projects"
      >
        Projects
      </a>
      <a
        href="test-file-stub"
      >
        Resume
      </a>
      <a
        href="#contact"
      >
        Contact
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
